.github/workflows/release-upload.yaml

```diff
@@ -0,0 +1,193 @@
+name: Release Upload
+# Pipeline to download binaries from Hydra builds and create GitHub
+# releases out of them (or attaching the binaries to an already existing release)
+#
+# This pipeline is divided in three jobs:
+#
+# 1. The "wait_for_hydra" job looks up the status of the Hydra build corresponding to
+#    the tag for which we want to release. If the status is success, the pipeline
+#    continues. If the status cannot be determined or is a failure, the pipeline stops.
+# 2. The "pull" job downloads the binaries from the Hydra build and uploads them as artifacts.
+#    This job uses a matrix to handle all 3 platforms. Uploading as an artifact allows to
+#    do the last job without a matrix.
+#
+#    This job uses `--builders "" --max-jobs 0` to ensure the assets are downloaded
+#    from the IOG cache.
+# 3. The "create_release" job downloads the artifacts and attaches them to the target release.
+#
+# This pipeline is triggered in one of the following 3 ways:
+#
+# 1. When a release is published, this release's commit must have a tag for the pipeline to succeed
+# 2. When a tag is pushed. If this tag has a corresponding release, the release will be augmented.
```

Comment:
What happens when you push a tag, then a release gets published and then you remove a tag and push it again (pointing to a different commit)? Will a release get overwritten?

Reply:
Good question. I don't know and I can't find the answer in the documentation of https://github.com/input-output-hk/action-gh-release. Let me try it out :+1:

Reply:
@carbolymer> the release is updated when the tag has changed between two executions of the pipeline :+1: 

See e.g. first release built:

![test-tag-1-release-1](https://github.com/IntersectMBO/cardano-cli/assets/634720/e541b165-50c3-457a-b25e-93ccc7604e21)

and then after the tag has been updated to another commit:

![test-tag-1-release-2](https://github.com/IntersectMBO/cardano-cli/assets/634720/4f15cc95-e86f-4882-b16b-aafa6488bfc7)

Reply:
Nice, seems fine!